README.md

```diff
@@ -136,122 +171,280 @@ Every resource in this repository is essential for the overall template setup. N
+<div class="container mx-auto px-4">        
+  <p> 
+    <h1>${apiName}</h1>
+    ${render(<ListChannels channels={channels} operationType='subscribe' />)}
+    ${render(<ListChannels channels={channels} operationType='publish' />)}
+    ${render(<DiagramContent asyncapi={asyncapi} />)}
+    ${render(<Extension asyncapi={asyncapi} />)}
+    ${render(<ExternalDocs asyncapi={asyncapi} />)}
+  </p> 
+</div>  
+`;
+}
+```
+
+#### File Templates
+
+The Generator has a feature called [`file templates`](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#file-templates) that allows you to create a template file with special that has `$$` markers, like `$$schema$$`. There are multiple different file templates available. In this Template, during generation, `$$schema$$` is replaced with a schema name, and Template gets the following variables in the context:
+
+- `schema` that is a Schema object map
+- `schemaName` that is the name of the schema
+
+Template file [template/schemas/$$schema$$.js](template/schemas/$$schema$$.js) is an example of such a file template:
+
+```js
+export default function({ schemaName, schema }) {
+  const name = normalizeSchemaName(schemaName);
+  return (
+    <File name={`${name}-example.html`}>
+      <SchemaFile schemaName={schemaName} schema={schema} />
+    </File>
+  );
+}
+
+function SchemaFile({ schemaName, schema }) {
+  // implementation...
+}
+```
 
-Checkout `template` directory to see how different features of the generator are presented.
+This one template file results in multiple HTML files, one per schema.
 
-#### Template Context
+### Nunjucks
+
+The list of resources that are relevant for this Template using Nunjucks:
+
+- `template_njk` is a directory where you keep all the files that will be processed by the Generator using Nunjucks. 
+- `partials_njk` is a directory where you keep [reusable parts](https://mozilla.github.io/nunjucks/templating.html#include) of the templates. You can also put there [Nunjucks macros](https://mozilla.github.io/nunjucks/templating.html#macro) that are like includes but customizable as you can pass custom arguments to them like to functions.
+- `filters` is a directory where you keep filters later used in files from the `template_njk` directory. Filters are normal JavaScript functions that you can apply to Nunjucks variables in the template files. It is [native Nunjucks functionality](https://mozilla.github.io/nunjucks/templating.html#filters), and we are just making it easier to provide them.
+
+> **NOTE**: If you are using Nunjucks as a rendering engine, you should delete the given folders which are related to React: `template`, `partials`. You should also change names of folders: `template_njk` -> `template` and `partials_njk` -> `partials`, remove the `renderer` field from [`generator`](#configuration) section and remove `react`, `@asyncapi/generator-react-sdk` and `source-map-support` dependencies in the `package.json` file under `dependencies` field.
```

Comment:
`source-map-support` and `react` are no longer required as dependencies for a react template
```suggestion
> **NOTE**: If you are using Nunjucks as a rendering engine, you should delete the given folders which are related to React: `template`, `partials`. You should also change names of folders: `template_njk` -> `template` and `partials_njk` -> `partials`, remove the `renderer` field from [`generator`](#configuration) section and remove `@asyncapi/generator-react-sdk` dependency in the `package.json` file under `dependencies` field.
```